packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -101,26 +101,29 @@ library LibProposing {
         // depend on Taiko DAO to make necessary adjustments to the rewards.
         uint256 reward;
 
-        if (config.proposerRewardPerSecond > 0 && config.proposerRewardMax > 0)
-        {
-            // Unchecked is safe as block.timestamp is always greater than
-            // block.proposedAt (proposed in the past)
-            unchecked {
-                uint256 blockTime = block.timestamp
-                    - state.blocks[(b.numBlocks - 1) % config.blockRingBufferSize]
-                        .proposedAt;
-
-                if (blockTime > 0) {
-                    reward = (config.proposerRewardPerSecond * blockTime).min(
-                        config.proposerRewardMax
-                    );
-
-                    // Reward must be minted
-                    LibTaikoToken.creditToken(
-                        state, resolver, msg.sender, reward, true
-                    );
-                }
+        // First proposer per L1 block updates the L1 block height and the accumulated reward.
+        if(state.slotC.latestRewardAccL1Height < block.number) {
+            // The very first proposal within the protocol starts the 'clock' and from that point on we calculate the accumulation.
+            if(state.slotC.latestRewardAccL1Height != 0) {
+                // Calculate the addition to the accumulated rewards by getting how many blocks elapsed since the last 'sync' then multiply it by 12 (L1 block time) and the reward issued per second.
+                state.slotC.accumulatedReward += uint128((block.number - state.slotC.latestRewardAccL1Height) * 12 * config.proposerRewardPerSecond);
```

Comment:
Lets use  `proposerReviewPerL1Block` to replace 12 * config.proposerRewardPerSecond`

Reply:
Assuming each Taiko block time is 3 seconds, then in each Etheruem block, there will be 4 L2 blocks. Their reward will be:

1/3, 2/9, 4/27, 8/81, or in percentage: 41.54%, 27.69%, 18.46%, and 12.31%,
with 3 blocks, 47.37%, 31.58%, and 21.05%;
with 2 blocks: 60% and 40% respectively.

Reply:
The ratios looks good